Use the same simulated iPhone model for layout tests an API tests

JonWBedard opened the pull request:
#### 4518454316d9cda0cb74ddcd7e757a57ff58feda
<pre>
Use the same simulated iPhone model for layout tests an API tests
<a href="https://bugs.webkit.org/show_bug.cgi?id=239885">https://bugs.webkit.org/show_bug.cgi?id=239885</a>
rdar://92505373

Rubber-stamped by Aakash Jain.

run-api-tests should use the same default device model that layout tests.

* Tools/Scripts/webkitpy/api_tests/manager.py:
(Manager._initialize_devices): Use the default device model as specified in the port
class. Add a temporary override for iOS until <a href="https://bugs.webkit.org/show_bug.cgi?id=256806">https://bugs.webkit.org/show_bug.cgi?id=256806</a>
is resolved.

Canonical link: <a href="https://commits.webkit.org/264115@main">https://commits.webkit.org/264115@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/783883a49460a4c9d7b17b9c5177edffa13b0bf7

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/6761 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/6977 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/7162 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/8349 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/7014 "Built successfully") 
| [  ~~🧪 bindings~~](https://ews-build.webkit.org/#/builders/9/builds/6759 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/8139 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/6922 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/9918 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/6878 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/15/builds/7472 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/6201 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/8444 "Built successfully") | 
| [  ~~🧪 webkitpy~~](https://ews-build.webkit.org/#/builders/6/builds/6865 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/39/builds/4906 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/10/builds/6121 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/13935 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/6649 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/36/builds/6201 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/8975 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/6689 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/17/builds/5483 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/30/builds/6075 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/4/builds/10252 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/784 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/6449 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

gsnedders commented on Tools/Scripts/webkitpy/api_tests/manager.py:
```diff
@@ -150,8 +151,11 @@ def _print_tests_result_with_status(self, status, runner):
+        # FIXME: Remove the special-case for iOS Simulator once https://bugs.webkit.org/show_bug.cgi?id=256806 is resolved
+        if self._port.port_name in ('iphone-simulator', 'ios-simulator'):
+            SimulatedDeviceManager.initialize_devices(DeviceRequest(DeviceType(hardware_family='iPhone', hardware_type='SE (3rd generation)'), allow_incomplete_match=True), self.host, simulator_ui=False)
+        elif 'simulator' in self._port.port_name:
```
Are there any cases where this differs to `.endswith('-simulator')`?

JonWBedard replied:
Not at the moment, but I think `in` more accurately reflects the sentiment here. If someone were to do something like `ios-simulator-legacy` (which would seem a perfectly reasonable port name), `ends with('-simulator')` here would break that class name.

aj062 reviewed (commented):
rs=me

JonWBedard committed:
Use the same simulated iPhone model for layout tests an API tests
https://bugs.webkit.org/show_bug.cgi?id=239885
rdar://92505373

Rubber-stamped by Aakash Jain.

run-api-tests should use the same default device model that layout tests.

* Tools/Scripts/webkitpy/api_tests/manager.py:
(Manager._initialize_devices): Use the default device model as specified in the port
class. Add a temporary override for iOS until https://bugs.webkit.org/show_bug.cgi?id=256806
is resolved.

Canonical link: https://commits.webkit.org/264115@main
